tests/network/pod_networking.go

```diff
@@ -0,0 +1,115 @@
+
+var _ = SIGDescribe("Pod Networking", func() {
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
+	})
+
+	vmiWithDefaultBinding := func() *v1.VirtualMachineInstance {
+		vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+		vmi.Spec.Domain.Devices.Interfaces = nil
+		vmi.Spec.Networks = nil
+		return vmi
+	}
+
+	vmiWithBridgeBinding := func() *v1.VirtualMachineInstance {
+		vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+		vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultBridgeNetworkInterface()}
+		vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+		return vmi
+	}
+
+	vmiWithMasqueradeBinding := func() *v1.VirtualMachineInstance {
+		vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+		vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultMasqueradeNetworkInterface()}
+		vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+		return vmi
+	}
+
+	vmiWithBinding := func(bindingMechanism string) *v1.VirtualMachineInstance {
+		switch bindingMechanism {
+		case "bridge":
+			return vmiWithBridgeBinding()
+		case "masquerade":
+			return vmiWithMasqueradeBinding()
+		case "default":
+			return vmiWithDefaultBinding()
+		default:
+			panic(fmt.Sprintf("Selected binding %q is invalid", bindingMechanism))
+		}
+	}
+
+	testStatusReporting := func(description string, bindingMechanism string) {
+		Context(fmt.Sprintf("VMI connected to the pod network using %s", description), func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				By("Creating the VMI")
+				var err error
+				vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmiWithBinding(bindingMechanism))
+				Expect(err).NotTo(HaveOccurred(), "VMI should be successfully created")
+
+				By("Waiting until the VMI gets ready")
+				vmi = tests.WaitUntilVMIReady(vmi, tests.LoggedInAlpineExpecter)
+			})
+
+			AfterEach(func() {
+				By("Deleting the VMI")
+				err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.GetName(), &metav1.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred(), "VMI should be successfully removed")
+			})
+
+			It("should report PodIP as its own on interface status", func() {
+				By("Getting pod of the VMI")
+				vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+
+				By("Making sure IP reported on the VMI matches the one on the pod")
+				Expect(vmi.Status.Interfaces[0].IP).To(Equal(vmiPod.Status.PodIP), "The address reported in VMI's IP should match the PodIP")
+				Expect(vmi.Status.Interfaces[0].IPs).NotTo(BeEmpty(), "VMI should report a list of its IP addresses")
+				Expect(vmi.Status.Interfaces[0].IPs[0]).To(Equal(vmiPod.Status.PodIP), "The first address reported in VMI's IPs should match the PodIP")
```

Comment:
Please drop the text in case of failure here, I think we can read this based on the assert params.
(there seem to be at least 3 times explaining what is tested, 1st by the "By", then by the variables and lastly by the error message.)

Reply:
It is redundant, but it documents the behavior to a casual reader who just goes over "strings". So one can just open the test and see that the report will show IP and IPs where the first one is IP, without getting into Expects. Unless it bothers you a lot, I'd prefer to keep it.

Reply:
You don't need to read the right side if you understood the left one and vice-versa. The text is meant for a casual reviewer who does not care about the test but rather about the behavior. For me, it increases the readability. I understand `"The first address reported in VMI's IPs should match the PodIP"` faster than when I'm parsing `Expect(vmi.Status.Interfaces[0].IPs[0]).To(Equal(vmiPod.Status.PodIP))`

Reply:
After I read it I understand that I do not need it.
The casual test reader better understand code, otherwise he should not read it.
Please do not overload the codebase with text. It is enough for me to suffer this BDD formatting, we do not need to go one step further and explain each an every assert textually when it is straight forward clear from the assert itself.

Reply:
I use the test suite as a test plan documenting in detail what is really the expected behavior. I'll drop the text to win your heart an approval though.

Reply:
My heart will thank you :)

Tests are indeed documentation, but not in the sense of human readable manual, but in the sense of examples of how to use the functionality and what to expect. BDD seems to take it one step further to read like a test plan from a book, which I still can live with (although still uncomfortable). But going into more details then that is a bit too much for me.